internal/job/githttptest/githttptest.go

```diff
@@ -0,0 +1,371 @@
+	}
+
+	commitCmd = exec.Command("git", "commit", "-m", "Add new file")
+	commitCmd.Dir = tempDir
+	if out, err := commitCmd.CombinedOutput(); err != nil {
+
+		return "", out, fmt.Errorf("failed to commit new file: %w", err)
+	}
+
+	remoteCmd := exec.Command("git", "remote", "set-url", "origin", s.RepoURL(repoName))
+	remoteCmd.Dir = tempDir
+	if out, err := remoteCmd.CombinedOutput(); err != nil {
+		return "", out, fmt.Errorf("failed to set remote URL: %w", err)
+	}
+
+	pushCmd := exec.Command("git", "push", "origin", branchName)
+	pushCmd.Dir = tempDir
+	if out, err := pushCmd.CombinedOutput(); err != nil {
+		return "", out, fmt.Errorf("failed to push branch %s: %w", branchName, err)
+	}
+
+	commitHashCmd := exec.Command("git", "rev-parse", branchName)
+	commitHashCmd.Dir = tempDir
+	commitHash, err := commitHashCmd.Output()
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to get commit hash: %w", err)
+	}
+
+	commitHashStr := strings.TrimSpace(string(commitHash))
+
+	return commitHashStr, nil, nil
+}
+
+func (s *Server) CreateRef(repoName, refName, commitHash string) (out []byte, err error) {
+	repoPath := filepath.Join(s.repositories, repoName)
+	// Check if repository exists
+	if _, err = os.Stat(repoPath); os.IsNotExist(err) {
+		return nil, fmt.Errorf("repository '%s' not found at path: %s", repoName, repoPath)
+	}
+
+	// git update-ref refs/heads/branch-name commit-sha
+	updateRefCmd := exec.Command("git", "update-ref", refName, commitHash)
+	updateRefCmd.Dir = repoPath
+	if out, err = updateRefCmd.CombinedOutput(); err != nil {
+		return nil, fmt.Errorf("failed to create ref %s: %w", refName, err)
+	}
+
+	return out, nil
+}
+
+func (s *Server) RepoURL(repoName string) string {
+	if s.URL == "" {
+		panic("Server not started")
+	}
+	return fmt.Sprintf("%s/%s.git", s.URL, repoName)
+}
+
+func (s *Server) Close() {
+	if s.Listener != nil {
+		s.Listener.Close()
+		s.wg.Wait()
+		s.Listener = nil
+	}
+	if s.repositories != "" {
+		os.RemoveAll(s.repositories)
+		s.repositories = ""
+	}
+	if s.Config != nil {
+		s.Config = nil
+	}
+	if s.URL != "" {
+		s.URL = ""
+	}
+}
+
+// isValidRepoName checks if the repository name is valid
+func isValidRepoName(name string) bool {
+	// Basic validation: non-empty, no slashes or special characters
+	// Note: .git suffix should already be removed before this check
+	return name != "" && !strings.ContainsAny(name, "\\/:*?\"<>|")
+}
+
+// initBareRepo initializes a new bare Git repository
+func initBareRepo(path string) error {
+	cmd := exec.Command("git", "init", "--bare", path)
+	return cmd.Run()
+}
+
+// handleGitUploadPack handles the git-upload-pack endpoint (used for git fetch/clone)
+func (s *Server) handleGitUploadPack(w http.ResponseWriter, r *http.Request) {
+	repoName := r.PathValue("repository")
+	repoName = strings.TrimSuffix(repoName, ".git")
+
+	if !isValidRepoName(repoName) {
+		w.WriteHeader(http.StatusBadRequest)
+		fmt.Fprintf(w, "Invalid repository name")
```

Comment:
This looks like a job for `http.Error`

```suggestion
		http.Error(w, "Invalid repository name", http.StatusBadRequest)
```